### ISSUE TITLE
Enable Three-finger click or Four-finger click   to Full Screen 

### ISSUE THREAD

**Issue by vincenty1ung** [aaronkollasch/jitouch#37]
Three-finger or Four-finger move will activate fullscreen，

**Comment by aaronkollasch** [comment]
I currently am using four-finger click to Full Screen, and it works fine. What is the issue you are describing?

**Comment by vincenty1ung** [comment]
> I currently am using four-finger click to Full Screen, and it works fine. What is the issue you are describing?

Is the system's native three-finger move  enabled?

**Comment by aaronkollasch** [comment]
Ah I just enabled the native three-finger move, and three-finger moves trigger the "Three-Finger Click" gesture. I think this is unavoidable, as the system is simulating a click while three fingers are on the trackpad, which looks exactly like a three-finger click. If you want to use the system's native three-finger move, I recommend disabling the "Three-Finger Click" gesture (as well as any "Three-Swipe-x" gestures if you haven't already).

With the system's three-finger move enabled, "Four-finger Click" still works fine for me, and I don't see an option for four-finger move. I have the native four-finger swipe set to switch desktops, and that doesn't interfere with "Four-Finger Click".

**Comment by vincenty1ung** [comment]
> Ah I just enabled the native three-finger move, and three-finger moves trigger the "Three-Finger Click" gesture. I think this is unavoidable, as the system is simulating a click while three fingers are on the trackpad, which looks exactly like a three-finger click. If you want to use the system's native three-finger move, I recommend disabling the "Three-Finger Click" gesture (as well as any "Three-Swipe-x" gestures if you haven't already).
> 
> With the system's three-finger move enabled, "Four-finger Click" still works fine for me, and I don't see an option for four-finger move. I have the native four-finger swipe set to switch desktops, and that doesn't interfere with "Four-Finger Click".

so，the system's native four-finger move enabled ，problem happend

**Comment by aaronkollasch** [comment]
Where are you enabling the system's native four-finger move?

**Comment by vincenty1ung** [comment]
> Where are you enabling the system's native four-finger move?
here
<img width="598" alt="image" src="https://user-images.githubusercontent.com/31126332/192423219-24c34478-8c18-498e-990c-af4f03d33394.png">
here

**Comment by aaronkollasch** [comment]
I have those gestures enabled, and Four-Finger Click works fine. Note that the system gestures are not compatible with any of the Four-Swipe gestures, however. You will have to remove any of Jitouch's Four-Swipe gestures.

<img width="668" alt="image" src="https://user-images.githubusercontent.com/7785912/192423495-c44f11f5-3fba-4b7f-b850-8258d8790d40.png">
<img width="668" alt="image" src="https://user-images.githubusercontent.com/7785912/192423584-83eb6959-aa95-4555-ac9d-818a4818d27c.png">

**Comment by vincenty1ung** [comment]
thanks ，i trying

> I have those gestures enabled, and Four-Finger Click works fine. Note that the system gestures are not compatible with any of the Four-Swipe gestures, however. You will have to remove any of Jitouch's Four-Swipe gestures.
> 
> <img alt="image" width="668" src="https://user-images.githubusercontent.com/7785912/192423495-c44f11f5-3fba-4b7f-b850-8258d8790d40.png">
> 
> <img alt="image" width="668" src="https://user-images.githubusercontent.com/7785912/192423584-83eb6959-aa95-4555-ac9d-818a4818d27c.png">

**Comment by aaronkollasch** [comment]
Hi, is the problem solved?

**Comment by vincenty1ung** [comment]
> Hi, is the problem solved?
Hi bro，
set  four-finger click to Full Screen
When typing, the left thumb is accidentally prevented from being in the upper left corner of the trackpad ，The left hand does not move for a long time, the thumb is on the touchpad in the upper left corner，three-finger move will activate fullscreen，

**Comment by vincenty1ung** [comment]
> Hi, is the problem solved?

Hi bro，
set four-finger click to Full Screen
When typing, the left thumb is accidentally prevented from being in the upper left corner of the trackpad ，The left hand does not move for a long time, the thumb is on the touchpad in the upper left corner，the other right hand three-finger move will activate fullscreen，

**Comment by aaronkollasch** [comment]
Yeah, looks like if a thumb is resting on the trackpad while performing a three-finger move, four-finger click gets activated. I'm not sure there's an easy fix for this other than enforcing that all four fingers have to be close together for a four-finger click. In the mean time, you could try disabling four-finger click and using four-finger tap for full-screen (in the [latest beta](https://github.com/aaronkollasch/jitouch/releases/tag/v2.82-beta.3)), or just try to lift all other fingers when performing a three-finger move.

**Comment by vincenty1ung** [comment]
> Yeah, looks like if a thumb is resting on the trackpad while performing a three-finger move, four-finger click gets activated. I'm not sure there's an easy fix for this other than enforcing that all four fingers have to be close together for a four-finger click. In the mean time, you could try disabling four-finger click and using four-finger tap for full-screen (in the [latest beta](https://github.com/aaronkollasch/jitouch/releases/tag/v2.82-beta.3)), or just try to lift all other fingers when performing a three-finger move.

 four-finger click ==  four-finger move Click Can't listen to the vibration event of the touchpad?

**Comment by vincenty1ung** [comment]
> Yeah, looks like if a thumb is resting on the trackpad while performing a three-finger move, four-finger click gets activated. I'm not sure there's an easy fix for this other than enforcing that all four fingers have to be close together for a four-finger click. In the mean time, you could try disabling four-finger click and using four-finger tap for full-screen (in the [latest beta](https://github.com/aaronkollasch/jitouch/releases/tag/v2.82-beta.3)), or just try to lift all other fingers when performing a three-finger move.

or just try to lift all other fingers when performing a three-finger move.
Reply：

**Comment by vincenty1ung** [comment]
> Yeah, looks like if a thumb is resting on the trackpad while performing a three-finger move, four-finger click gets activated. I'm not sure there's an easy fix for this other than enforcing that all four fingers have to be close together for a four-finger click. In the mean time, you could try disabling four-finger click and using four-finger tap for full-screen (in the [latest beta](https://github.com/aaronkollasch/jitouch/releases/tag/v2.82-beta.3)), or just try to lift all other fingers when performing a three-finger move.

or just try to lift all other fingers when performing a three-finger move.
Reply：This is not a special operation by myself, it is accidentally triggered when typing, and the probability of occurrence is very high.

**Comment by aaronkollasch** [comment]
What trackpad gestures do you have enabled in Jitouch? Can you send screenshots?
Also, I don't understand the problem you have when typing. Is it that one palm is resting on the trackpad while doing a three-finger move? From your description it sounds like you had extra fingers on the trackpad while performing a three-finger move.

**Comment by vincenty1ung** [comment]
> What trackpad gestures do you have enabled in Jitouch? Can you send screenshots?
> Also, I don't understand the problem you have when typing. Is it that one palm is resting on the trackpad while doing a three-finger move? From your description it sounds like you had extra fingers on the trackpad while performing a three-finger move.

ok
<img width="780" alt="image" src="https://user-images.githubusercontent.com/31126332/194737193-7802eb15-10aa-4f52-a416-ba7ad62a1829.png">

hello is this
<img width="605" alt="image" src="https://user-images.githubusercontent.com/31126332/194737389-09d62e2f-973d-4bc8-b878-6695067ab7b9.png">

**Comment by aaronkollasch** [comment]
Ok, thanks - I think the solution to this will be configurable deadzones at each edge of the trackpad where fingers are not detected. There is a built in system for palm rejection that doesn't work well enough with the newer trackpads. I will see about improving the palm rejection when I can.

Related to #18

**Comment by vincenty1ung** [comment]
> Ok, thanks - I think the solution to this will be configurable deadzones at each edge of the trackpad where fingers are not detected. There is a built in system for palm rejection that doesn't work well enough with the newer trackpads. I will see about improving the palm rejection when I can.
> 
> Related to #18

thanks hard work
